File: TrainerAI/AITrainer-Backend/AITrainer/AITrainer.DomainModel/Models/LeaveApplication.cs
namespace AITrainer.AITrainer.DomainModel.Models
{
    public class LeaveApplication
    {
        public string Id { get; set; }
        public string InternId { get; set; }
        public DateTime leaveStartDate { get; set; }
        public DateTime? leaveEndDate { get; set; }
        public int? totalDay { get; set; }
        public string leaveType { get; set; }
        public string? leaveReason { get; set; }
        public string leaveStatus { get; set; }
        public string leaveCategory { get; set; }
        public bool isDeleted { get; set; }
        public string? approvedBy { get; set; }
        public DateTime? approvedDate { get; set; }
        public string? Comments { get; set; }
        public string createdBy { get; set; }
        public string updatedBy { get; set; }
        public DateTime createdDate { get; set; }
        public DateTime updatedDate { get; set; }
        public List<LeaveDocumentAttachment>? Attachments { get; set; } = new List<LeaveDocumentAttachment>();


    }
    public class LeaveDocumentAttachment
    {
        public string Id { get; set; }
        public string FileName { get; set; }
        public byte[] FileData { get; set; }
        public string? Description { get; set; }
        public bool IsDeleted { get; set; } = false;
        public LeaveApplication leaveApplication { get; set; }
        public string LeaveApplicationId { get; set; }
    }
}

File: TrainerAI/AITrainer-Frontend/src/app/model/LeaveApplication.ts
import { imagsDetails } from "./feedback";

export interface applyLeave {
    leaveStartDate: Date,
    leaveEndDate: Date,
    leaveType: string,
    leaveReason: string
}
export interface leaveApproval {
    leaveStartDate: Date;
    name: string;
    leaveReason: string;
    totalDay: number;
    leaveEndDate: Date;
}
export enum LeaveType {
    SickLeave = 'Sick Leave',
    CasualLeave = 'Casual Leave',
    WorkFromHome = 'Work From Home',

}
export enum LeaveCategory {
    FirstHalf = 'First Half',
    SecondHalf = 'Second Half',
}
export interface leaveApplication {
    internId: string;
    leaveStartDate: Date;
    leaveEndDate: Date;
    totalDay: number;
    leaveType: string;
    leaveReason: string;
    leaveStatus: string;
    isDeleted: boolean;
    approvedBy: string;
    approvedDate: Date;
    comments: string;
    createdBy: string;
    updatedBy: string;
    createdDate: Date;
    updatedDate: Date;
    attachments: LeaveDocumentAttachmentDto
}
export interface AdminLeaveView {
    id: string;
    internId: string;
    name: string;
    leaveStartDate: Date;
    leaveEndDate: Date;
    totalDay: number;
    leaveType: string;
    leaveReason: string;
    leaveStatus: string;
    updatedBy: string;
    createdDate: string;
}
export interface ListAdminAttendanceView {
    leaveApplication: AdminLeaveView[];
}
export interface LeaveDocumentAttachmentDto {
    id: string;
    fileName: string;
    fileData: string;
}